contracts/governance/src/ZeroExVotes.sol

```diff
@@ -0,0 +1,282 @@
+     */
+    function checkpoints(address account, uint32 pos) public view override returns (Checkpoint memory) {
+        return _checkpoints[account][pos];
+    }
+
+    /**
+     * @inheritdoc IZeroExVotes
+     */
+    function numCheckpoints(address account) public view override returns (uint32) {
+        return SafeCast.toUint32(_checkpoints[account].length);
+    }
+
+    /**
+     * @inheritdoc IZeroExVotes
+     */
+    function getVotes(address account) public view override returns (uint256) {
+        uint256 pos = _checkpoints[account].length;
+        return pos == 0 ? 0 : _checkpoints[account][pos - 1].votes;
+    }
+
+    /**
+     * @inheritdoc IZeroExVotes
+     */
+    function getQuadraticVotes(address account) public view override returns (uint256) {
+        uint256 pos = _checkpoints[account].length;
+        return pos == 0 ? 0 : _checkpoints[account][pos - 1].quadraticVotes;
+    }
+
+    /**
+     * @inheritdoc IZeroExVotes
+     */
+    function getPastVotes(address account, uint256 blockNumber) public view override returns (uint256) {
+        require(blockNumber < block.number, "ZeroExVotes: block not yet mined");
+
+        Checkpoint memory checkpoint = _checkpointsLookup(_checkpoints[account], blockNumber);
+        return checkpoint.votes;
+    }
+
+    /**
+     * @inheritdoc IZeroExVotes
+     */
+    function getPastQuadraticVotes(address account, uint256 blockNumber) public view override returns (uint256) {
+        require(blockNumber < block.number, "ZeroExVotes: block not yet mined");
+
+        Checkpoint memory checkpoint = _checkpointsLookup(_checkpoints[account], blockNumber);
+        return checkpoint.quadraticVotes;
+    }
+
+    /**
+     * @inheritdoc IZeroExVotes
+     */
+    function getPastTotalSupply(uint256 blockNumber) public view override returns (uint256) {
+        require(blockNumber < block.number, "ZeroExVotes: block not yet mined");
+
+        // Note that due to the disabled updates of `_totalSupplyCheckpoints` in `writeCheckpointTotalSupply` function
+        // this always returns 0.
+        Checkpoint memory checkpoint = _checkpointsLookup(_totalSupplyCheckpoints, blockNumber);
+        return checkpoint.votes;
+    }
+
+    /**
+     * @inheritdoc IZeroExVotes
+     */
+    function getPastQuadraticTotalSupply(uint256 blockNumber) public view override returns (uint256) {
+        require(blockNumber < block.number, "ZeroExVotes: block not yet mined");
+
+        // Note that due to the disabled updates of `_totalSupplyCheckpoints` in `writeCheckpointTotalSupply` function
+        // this always returns 0.
+        Checkpoint memory checkpoint = _checkpointsLookup(_totalSupplyCheckpoints, blockNumber);
+        return checkpoint.quadraticVotes;
+    }
+
+    /**
+     * @inheritdoc IZeroExVotes
+     */
+    function moveVotingPower(address src, address dst, uint256 srcBalance, uint256 dstBalance, uint256 amount) public {
+        if (src != dst && amount > 0) {
```

Comment:
Although this seems like an optimization, it will make a future migration harder to apply.
```suggestion
        if (src != dst) {
```